### ISSUE TITLE
Won't open mobile.twitter.com in container.

### ISSUE THREAD

**Issue by olliebean** [kintesh/containerise#25]
The rule is "*.twitter.com , Twitter". It doesn't open mobile.twitter.com in the Twitter container. Even explicitly adding "mobile.twitter.com , Twitter" as a rule doesn't work. Other mobile sites, e.g. mobile.facebook.com, seem to work fine.

**Comment by kintesh** [comment]
Works for me fine. Make sure you have "Twitter" container before adding the rule and that you are not using anyother addon that uses containers api.

**Comment by olliebean** [comment]
The Twitter container was created before adding the rule. Incidentally, "twitter.com" opens in the Twitter container as expected - it's only "mobile.twitter.com" that doesn't.

The only other container-using addon I'm using is Context Plus, but the issue persists when I disable that addon.

**Comment by olliebean** [comment]
The Twitter container was created before adding the rule. Incidentally, "twitter.com" opens in the Twitter container as expected - it's only "mobile.twitter.com" that doesn't.

The only other container-using addon I'm using is Context Plus, but the issue persists when I disable that addon.

I also tried deleting the rule and the container, then re-adding the container and the rule in that order, but the issue still persists.

**Comment by grahamperrin** [comment]
Works for me. Containerise 2.3.0 with Waterfox 56.2.0. 

```text
$ pkg info waterfox | grep Version
Version        : 56.2.0.31
$ date ; uname -v
Fri  1 Jun 2018 19:07:41 BST
FreeBSD 12.0-CURRENT #3 r334382: Thu May 31 01:37:12 BST 2018     root@momh167-gjp4-hpelitebook8570p-freebsd:/usr/obj/usr/src/amd64.amd64/sys/GENERIC 
```

My `storage.js` comprises: 

`{"map=mobile.twitter.com":{"host":"mobile.twitter.com","containerName":"Twitter","cookieStoreId":"firefox-container-38","enabled":true},"map=twitter.com":{"host":"twitter.com","containerName":"Twitter","cookieStoreId":"firefox-container-38","enabled":true}}`

**Comment by grahamperrin** [comment]
I reduced mine (in the GUI) to one rule instead of two, 

`*.twitter.com`

For me, this catches both domains.

**Comment by olliebean** [comment]
Behaving very strangely for me - if I open mobile.twitter.com in an existing tab which is already in a different container, it correctly switches to the Twitter container. But if I open it in a new tab, or in a tab that isn't in a container, it doesn't switch and stays non-containerised.

**Comment by grahamperrin** [comment]
Let's double-check: 

> … The only other container-using addon I'm using is Context Plus, …

– meaning, you do **not** use Mozilla's [Firefox Multi-Account Containers](https://addons.mozilla.org/addon/multi-account-containers/) extension. Please, do I understand correctly?

**Comment by olliebean** [comment]
No, I don't use the Firefox Multi-Account Containers extension, as I find it easier to assign containers with Containerise. Also I've had conflicts between the two extensions when I had them both installed.

**Comment by olliebean** [comment]
OS: Windows 10
Firefox: 60.0.1 (64-bit)

**Comment by grahamperrin** [comment]
Thanks, 

> … conflicts between the two extensions …

Exactly. 

I asked because here, after previously finding no problem with Containerise: 

1. Containerise ceased working (for mobile.twitter.com) when I enabled Firefox Multi-Account Containers
2. I also found Mozilla's extension not working as expected – I simply could not get the GUI with either (a) outdated version 4.1.0, which supposedly **is** compatible with (Firefox 56-based) Waterfox or (b) current version 6.0.0, which supposedly is **not** compatible with Waterfox
3. Containerise worked again immediately after I disabled Mozilla's extension. 

Eventually I removed Mozilla's extension. 

----

I now have Containerise 2.4.0 and Firefox Multi-Account Containers 6.0.0 enabled concurrently with Waterfox. Whilst I must not expect full compatibility, I do have: 

- a working GUI for Mozilla's extension
- Containerise still working for mobile.twitter.com

– **plus** a notion that the prior set of symptoms (basic functional failures of both Firefox Multi-Account Containers and Containerise) had a common cause that was worked around, hopefully resolved, by removal then (re)installation of Mozilla's extension. 

## Suggestion

@olliebean 

1. exit Firefox
2. backup your profile (or at least, `containers.json` and any other file that may be containers-related). 

Then it may be useful to tell what happens in response to you doing pretty much the same as I did a day or so ago: 

1. start Firefox 
2. remove Firefox Multi-Account Containers
3. exit Firefox
4. start Firefox
5. install Firefox Multi-Account Containers
6. test whether mobile.twitter.com is contained as it should be
7. disable Firefox Multi-Account Containers.

**Comment by olliebean** [comment]
Sorry, no, that didn't work. Containerise is still switching to the Twitter container for mobile.twitter.com if and only if it's opened in a tab that is already in another container.

What does work is installing Containerise to a fresh profile, and importing my rules from my existing profile. So there must be something else in my profile, perhaps a clash with another extension, that is causing it.

**Comment by grahamperrin** [comment]
OK, can you share troubleshooting information? Text (not raw data) copied from: 

- about:support

Thanks

**Comment by olliebean** [comment]
Here's my troubleshooting info:

[troubleshooting.txt](https://github.com/kintesh/containerise/files/2075633/troubleshooting.txt)

**Comment by grahamperrin** [comment]
Thanks. 

At the time of writing, with (lesser used) Firefox, symptoms are evident with Firefox Multi-Account Containers disabled. 

I renamed an old container to Twitter, then used Containerise to specify `*.twitter.com` for the container. 

<kbd>Control</kbd>-<kbd>T</kbd> followed by manually going to `twitter.com/` (auto-completed): success. 

<kbd>Control</kbd>-<kbd>T</kbd> followed by manually going to `mobile.twitter.com/` (auto-completed): failure; no containment. 

[Switch Container Plus](https://addons.mozilla.org/addon/switch-container-plus/reviews/) 0.14 enabled. Whilst I don't imagine a conflict in the Firefox environment, it _is_ bugged with Waterfox so I disabled Switch Container Plus, tested, enabled [Switch Container](https://addons.mozilla.org/addon/switch-container/), tested. Test results: 

- failure
- failure.

Conex enabled, I relaxed its preferences to enable moving tabs between containers. Test result: 

- failure. 

For privacy/security, I reverted the Conex preference. 

A case such as this should be an ideal candidate for [Extension Conflict Troubleshooter](https://addons.mozilla.org/addon/extension-conflict-troubleshoo/) a.k.a. FECT but Firefox Quantum disallows such things 👎 

My `containers.json`: 

`{"version":3,"lastUserContextId":34,"identities":[{"userContextId":5,"public":false,"icon":"","color":"","name":"userContextIdInternal.thumbnail","accessKey":""},{"userContextId":19,"public":true,"icon":"circle","color":"purple","name":"FreshPorts"},{"userContextId":31,"public":true,"icon":"circle","color":"red","name":"FreeBSD"},{"userContextId":32,"public":true,"icon":"circle","color":"green","name":"Mozilla"},{"userContextId":33,"public":true,"icon":"circle","color":"blue","name":"Twitter"},{"userContextId":34,"public":true,"icon":"fingerprint","color":"blue","name":"five"}]}`

– and troubleshooting information: 

- [2018-06-08 04-40 Firefox 60.0.1, everyday.txt](https://github.com/kintesh/containerise/files/2082854/2018-06-08.04-40.Firefox.60.0.1.everyday.txt)

**Comment by grahamperrin** [comment]
In my Firefox case the simplest workaround seems to be: 

- disallow multi-process (e10s). 

### Steps to workaround

1. about:config?filter=browser.tabs.remote.force-disable
2. if the boolean `browser.tabs.remote.force-disable` does not exist, create it
3. make it `true`
4. quit
5. start Firefox.

@olliebean please, is that a workaround for your case?

----

@kintesh whilst I don't expect users to appreciate the benefits of single-process Firefox Quantum (a blend of pros and cons), the apparent trouble in multi-process mode might give you food for thought towards a fix for this issue. 

Thanks

**Comment by olliebean** [comment]
Yes, creating `browser.tabs.remote.force-disable` and setting it to `true` does fix the problem for me. As for the pros and cons of single-process Quantum, I'll leave it like that for a bit and see how it goes.

**Comment by olliebean** [comment]
Seems the main pro of single-process is the ability to containerise mobile.twitter.com reliably, and the main con is Firefox runs noticeably laggier. I think I'll put up with the Twitter issue for now.
